core/crypto/signed_hdr.c

```diff
@@ -59,46 +60,56 @@ TEE_Result shdr_verify_signature(const struct shdr *shdr)
 	struct rsa_public_key key = { };
 	TEE_Result res = TEE_SUCCESS;
 	uint32_t e = TEE_U32_TO_BIG_ENDIAN(ta_pub_key_exponent);
+	struct ftmn ftmn = { };
+	unsigned int err_incr = 2;
 	size_t hash_size = 0;
 	size_t hash_algo = 0;
 
 	if (shdr->magic != SHDR_MAGIC)
-		return TEE_ERROR_SECURITY;
+		goto err;
```

Comment:
I changed this to 'goto err' on OP-TEE 3.14 and ran 'xtest -t regression 1008' and encountered an 'unhandled pageable abort' error. After analysis, I think it is because crypto_acipher_alloc_rsa_public_key is not executed but crypto_acipher_free_rsa_public_key is executed. Not sure if the latest version is OK.

Reply:
Looking at the (latest) implementation, I think it currently works fine as `key` is 0 initialized and all public key free possible callback handles (ltc, mbedtls, versal, caam and se050 do behave nicely in such context. Yet, I may have missed something and all in one, it looks a bit fragile and I think you're right, `shdr_verify_signature()` exit error sequence should not always call `crypto_acipher_free_rsa_public_key()`.

Reply:
The error is really because the key was not initialized to 0 on the old version. After initialization, all xtests passed. Thanks!

Reply:
For info: looking into `struct rsa_keypair` management, few other places assume it is safe to call `crypto_acipher_free_rsa_public_key()` on a key ref that was zero initialized and not allocated or partially allocated, for example `sw_crypto_acipher_alloc_rsa_keypair()` in _lib/libtomcrypt/rsa.c_ and _lib/libmbedtls/core/rsa.c_. So contrary to above my comment, current latest implementation is fine.